### PR TITLE
feat: support depth URL argument in Terraform modules

### PR DESCRIPTION
**Pull request by Moglum**
## Changes

Add Support for renovating Terraform Modules from Git repositories using shallow clone URL argument by expanding the Regex

## Context

Described in here: https://github.com/renovatebot/renovate/discussions/21266

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/terraform/__fixtures__/modules.tf [hunk 1]
```diff
@@ -62,7 +62,7 @@ module "task_definition" {
   source = "github.com/tieto-cem/terraform-aws-ecs-task-definition?ref=v0.1.0"
   name   = "mytask"
   container_definitions = [
-    "${module.container_definition.json}"]
+  "${module.container_definition.json}"]
```
Comment (viceice) [inline lib/modules/manager/terraform/__fixtures__/modules.tf:65]: avoid unrelated changes

File: lib/modules/manager/terraform/__fixtures__/modules.tf [hunk 2]
```diff
@@ -135,3 +135,7 @@ module "relative" {
 module "nosauce" {
   foo = "bar"
 }
+
+module "ssh_foo" {
+  source = "git::ssh://git@git.example.com/modules/foo-module.git//bar?depth=1&ref=v1.2.3"
```
Comment (viceice) [inline lib/modules/manager/terraform/__fixtures__/modules.tf:140]: this test can be removed. testing the regex below is enough

**Review by viceice — COMMENTED** [review]
shouldn't we update the other regex too?

**Comment by Moglum** [comment]
> shouldn't we update the other regex too?

Great idea. Updated all the other git regexes as well.

**Review by secustor — DISMISSED** [review]

File: lib/modules/manager/terraform/extractors/others/modules.ts [hunk 3]
```diff
@@ -10,16 +10,16 @@ import { DependencyExtractor } from '../../base';
 import type { TerraformDefinitionFile } from '../../hcl/types';
 
 export const githubRefMatchRegex = regEx(
-  /github\.com([/:])(?<project>[^/]+\/[a-z0-9-_.]+).*\?ref=(?<tag>.*)$/i
+  /github\.com([/:])(?<project>[^/]+\/[a-z0-9-_.]+).*\?.*ref=(?<tag>.*)$/i
```
Comment (viceice) [inline lib/modules/manager/terraform/extractors/others/modules.ts:13]: what of the depth and ref are reversed? then all is captured as ref. we should probably tighten the regex some more to allow any param order.

Comment (Moglum) [reply lib/modules/manager/terraform/extractors/others/modules.ts:13]: My reasoning was that you always want the `ref` at the end for readability, but you're right. It's still a valid URL even with `ref` before `depth`

**Review by secustor — APPROVED** [review]

**Review by viceice — APPROVED** [review]
